src/ucp/wireup/select.c

```diff
@@ -1678,6 +1696,8 @@ ucp_wireup_add_am_bw_lanes(const ucp_wireup_select_params_t *select_params,
     ucp_context_h context  = ep->worker->context;
     unsigned ep_init_flags = ucp_wireup_ep_init_flags(select_params,
                                                       select_ctx);
+    ucp_wireup_dev_usage_count dev_count = {};
```

Comment:
We might have to track `dev_count` per memory type. Previously we were starting with `dev_count` local and remote equal to zero for each mem type.
Now we can have host local from 0 to 2, cuda from 2 to 4, local from 4 to 6...

Reply:
I asked about this before and @brminich said it should be ok https://github.com/openucx/ucx/pull/9652#issuecomment-1920994559

Reply:
after reading comment, I synced with @brminich.

Reply:
@tvegas1 I made changes to track `dev_count` per memory type because otherwise, `am_lane` exclusion for one memory type will incorrectly increase `dev_count` for every other memory type.
Also, I had to set `am_lane` to `UCP_NULL_LANE` after the first iteration of rma_bw lanes addition. Otherwise, `dev_count` will be wrongfully incremented in each iteration.